src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -934,13 +969,17 @@ void LocalCleanupContainer(object container)
 
 					LocalCleanupContainer(container);
 					RequestLayoutPartial();
+					SaveContainersForIndexRepair(args.OldStartingIndex, -args.OldItems.Count);
```

Comment:
The way this method is used in ListViewBase is to call it _before_ actually applying the collection change, to save the prior state. Here that would mean calling it before `ItemsPanel.Children.RemoveAt`.

Reply:
Resolved according your suggestion.